### DISCUSSION TITLE
Color emojis look small

### DISCUSSION THREAD

**Discussion by VPraharsha03** [ryanoasis/nerd-fonts#1495] [Q&A] [answered]
Prompt: Starship
Font: JetBrains Mono Nerd Font

![image](https://github.com/ryanoasis/nerd-fonts/assets/56556436/68fd9dff-5d89-4067-905d-b8c243b42fcd)

What could be the reason that the clock color emoji right after "took" looks so small ?

**Comment by Finii** [comment, marked as answer]
Color emoji are not in any Nerd Font, so it comes from another font like `NotoColorEmoji.ttf`.

What happens then is depending on your terminal. It looks like your terminal scales foreign icons down to fit into one cell if they are too big. Which is a common strategy. The icons in the `Nerd Font` font are on the other hand scaled to be ~1.6 cells wide but they pretend to be exactly one cell wide, look the Windows icon, that is too wide for one cell.

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/a577a4d2-4382-4bbb-a366-664746314dd7)


You also see some 'not so smart' scaling is going on with the color-clock, as it is far too much down.

I do not know of any terminal that allows to scale foreign icons bigger than one cell, and of none that scales them down more intelligently like 'keep centered'.

The only solution I can suggest is: Use a Nerd Font icon of a clock instead - but that would be monochrome then.
Examples (there are a lot more, use the cheat sheet yourself...):

![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/48598aef-4db0-4ec9-b02d-82caf33c2f57)

**Reply by VPraharsha03** [reply]
@Finii the terminal i'm using is Windows Terminal (sorry should've mentioned before) and thanks a lot for the detailed explanation, and yes i thought of the same approach of using monochrome symbols as a quick fix but just opened this discussion just to know of anyother solution for this..

also doesn't windows use `SegoeUI` to take it's emojis from ?

**Reply by VPraharsha03** [reply]
> I do not know of any terminal that allows to scale foreign icons bigger than one cell, and of none that scales them down more intelligently like 'keep centered'.

i'm interested to know how do nerd font icons pretend to be of one cell or why are icons from same font get scaled to 1.6x ? what if nerdfonts had color emojis in them, will they be largely scaled in a similar way ?

**Reply by Finii** [reply]
> also doesn't windows use SegoeUI to take it's emojis from ?

I have no clue. I would doubt that.

> know how do nerd font icons pretend to be of one cell

The fonts in reality do not have the concept of a "cell". The visible stroke of a glyph and where the cursor starts and how much it advances is not fixed together. The bigger icons have just an "advance width" that is the same as all letters, so the next letter would overwrite the previous one. See maybe here
* #969 

> why are icons from same font get scaled to 1.6x 

In fact they are not scaled but the original size is kept. That varies, depending on text-font and individualk glyph set, between maybe 1.1 and 1.8 widths.

> will they be largely scaled in a similar way ?

Of course ;-)

But the Nerd Fonts patcher is based on Fontforge, and Fontforge is not really able to handle color glyphs. There are also other problems with color glyphs and I am not sure we will ever add them; but first we would need to change the underlying toolset, i.e. leave Fontforge and use fonttools for example. Which is a major rewrite.
